cpu/qn908x/Kconfig

```diff
@@ -50,4 +50,13 @@ config HAS_CPU_QN908X
 # Other cpu configuration
 rsource "Kconfig.clk"
 
+# ADC configuration
+config BOARD_HAS_ADC_PA06_CAP
+    bool
+    help
+         Indicates that the board has an external capacitor connected to PA06
+         which shold be used to stabilize the ADC reference voltage.
+         This external capacitor is optional but it can reduce the noise of the
+         ADC sampling.
```

Comment:
I'm not sure if Kconfig is the right way to expose this, as this is a hardware property that will not change / is not user configurable.

I'd rather suggest an ordinary `define` like `NRF5X_ENABLE_DCDC` / `USE_VREG_BUCK`.

This documentation would then go inside a 

```C
#if DOXYGEN
/**
 * @brief Indicates that the board has an external capacitor connected to PA06
 *        which shold be used to stabilize the ADC reference voltage.
 *        This external capacitor is optional but it can reduce the noise of the
 *        ADC sampling.
 */
#define BOARD_HAS_ADC_PA06_CAP
#endif
```

block in the cpu header.

@leandrolanzieri what do you think?

Reply:
Note that this block doesn't have a `prompt`, only a config so this doesn't show up in the user menu but it can be defined from the board's Kconfig. I did something similar for the clocks (whether there's a crystal on the board or not) although in that case there were Kconfig user prompts that depend on some of those settings.

Reply:
Ah does that work already? AFAIK Kconfig is not evaluated by default yet, only if a `.config` file exists as we are still in the transition period and not everything is on Kconfig yet.

However, I don't follow the developments there closely.

Reply:
It looks like it does. I just tried:
```
rm tests/periph_adc/bin -r
make BOARD=qn9080dk -C tests/periph_adc
grep BOARD_HAS_XTAL32K tests/periph_adc -r
```
and the macro shows up in `tests/periph_adc/bin/qn9080dk/riotbuild/riotbuild.h`.

I honestly don't know if Kconfig or periph_cpu.h is the preferred way for this sort of config, but the "BOARD_HAS_..." macros where here so I added it here.

Reply:
If not enabled explicitly (`SHOULD_RUN_KCONFIG=1`) Kconfig is not executed, that's why the symbol is not coming from `out.config`. We have been using this prompt-less symbols to enable certain configurations (like clocks) inside Kconfig. But if this is not the case you should just define it in the header file as @benpicco pointed out, so the symbol exists even when Kconfig is not used.

Reply:
This is still pending, I'll do it once I get a chance.